Subject: [PATCH] Avoid add null name or icon to the chosen list

If an exception is thrown when getting icon or name, icon and/or name
will be null. We should not add null name or icon to the list.

diff --git a/core/java/android/accounts/ChooseAccountTypeActivity.java b/core/java/android/accounts/ChooseAccountTypeActivity.java
index a3222d8..81607d8 100644
--- a/core/java/android/accounts/ChooseAccountTypeActivity.java
+++ b/core/java/android/accounts/ChooseAccountTypeActivity.java
@@ -146,8 +146,10 @@
                     Log.w(TAG, "No icon resource for account type " + desc.type);
                 }
             }
-            AuthInfo authInfo = new AuthInfo(desc, name, icon);
-            mTypeToAuthenticatorInfo.put(desc.type, authInfo);
+            if (name != null && icon != null) {
+                AuthInfo authInfo = new AuthInfo(desc, name, icon);
+                mTypeToAuthenticatorInfo.put(desc.type, authInfo);
+            }
         }
     }
 
